marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/map-multiple-entities/writeEntityInstances.sjs

```diff
@@ -0,0 +1,42 @@
+const DataHubSingleton = require("/data-hub/5/datahub-singleton.sjs");
+const test = require("/test/test-helper.xqy");
+const datahub = DataHubSingleton.instance();
+
+const content = ["/content/customerInfo.json"].map(uri => {
+  return {
+    uri: uri,
+    value: cts.doc(uri)
+  };
+});
+
+let results = datahub.flow.runFlow('customerFlow', 'test-job', content, {outputFormat: 'json', mapping:{name:'mapCustomersJSON'}}, 1);
+let customerUris = getUris("Customer");
+let orderUris = getUris("Order");
+let productUris = getUris("Product");
+
+let assertions = [
+  test.assertEqual(1, results.totalCount),
+  test.assertEqual(0, results.errorCount),
+  test.assertEqual(1, results.completedItems.length),
+  test.assertEqual(1, customerUris.length),
+  test.assertEqual("/content/customerInfo.json", customerUris[0]),
+  test.assertEqual(2, orderUris.length),
+  test.assertEqual(3, productUris.length),
+  /* Not checking for permissions of the main entity instance since we have tests for it and also permissions passed through options
+  won't get applied before runFlow() is called. */
+  test.assertEqual("data-hub-operator,read,data-hub-operator,update", getDocPermissions(productUris[0])),
+  test.assertEqual("data-hub-common,read,data-hub-common,update", getDocPermissions(orderUris[0]))
+];
+
+function getUris(collection){
+  return fn.head(xdmp.eval('cts.uris(null, null, cts.collectionQuery("'+ collection+ '")).toArray()'))
+}
+
+function getDocPermissions(uri){
```

Comment:
Use getRecord in data-hub-test-helper.sjs, it gets you all this data in an easy-to-consume way.